credentials/cache/file.go

```diff
@@ -0,0 +1,204 @@
+	}
+	// verify the cache is working
+	if _, err := c.load(); err != nil {
+		return nil, fmt.Errorf("load: %w", err)
+	}
+	return c, nil
+}
+
+// Store implements the TokenCache interface.
+func (c *FileTokenCache) Store(key string, t *oauth2.Token) error {
+	c.locker.Lock()
+	defer c.locker.Unlock()
+	f, err := c.load()
+	if err != nil {
+		return fmt.Errorf("load: %w", err)
+	}
+	if f.Tokens == nil {
+		f.Tokens = map[string]*oauth2.Token{}
+	}
+	f.Tokens[key] = t
+	raw, err := json.MarshalIndent(f, "", "  ")
+	if err != nil {
+		return fmt.Errorf("marshal: %w", err)
+	}
+	return os.WriteFile(c.fileLocation, raw, ownerReadWrite)
+}
+
+// Lookup implements the TokenCache interface.
+func (c *FileTokenCache) Lookup(key string) (*oauth2.Token, error) {
+	c.locker.Lock()
+	defer c.locker.Unlock()
+	f, err := c.load()
+	if err != nil {
+		return nil, fmt.Errorf("load: %w", err)
+	}
+	t, ok := f.Tokens[key]
+	if !ok {
+		return nil, ErrNotConfigured
+	}
+	return t, nil
+}
+
+// init initializes the token cache file. It creates the file and directory if
+// they do not already exist.
+func (c *FileTokenCache) init() error {
+	// set the default file location
+	if c.fileLocation == "" {
+		home, err := os.UserHomeDir()
+		if err != nil {
+			return fmt.Errorf("failed loading home directory: %w", err)
+		}
+		c.fileLocation = filepath.Join(home, tokenCacheFilePath)
+	}
+	// create the directory if it doesn't already exist
+	if _, err := os.Stat(filepath.Dir(c.fileLocation)); err != nil {
+		if !os.IsNotExist(err) {
+			return fmt.Errorf("stat directory: %w", err)
+		}
+		// create the directory
+		if err := os.MkdirAll(filepath.Dir(c.fileLocation), ownerExecReadWrite); err != nil {
+			return fmt.Errorf("mkdir: %w", err)
+		}
+	}
+	// create the file if it doesn't already exist
+	if _, err := os.Stat(c.fileLocation); err != nil {
+		if !os.IsNotExist(err) {
+			return fmt.Errorf("stat file: %w", err)
+		}
+		f := &tokenCacheFile{
+			Version: tokenCacheVersion,
+			Tokens:  map[string]*oauth2.Token{},
+		}
+		raw, err := json.MarshalIndent(f, "", "  ")
+		if err != nil {
+			return fmt.Errorf("marshal: %w", err)
+		}
+		if err := os.WriteFile(c.fileLocation, raw, ownerReadWrite); err != nil {
+			return fmt.Errorf("write: %w", err)
+		}
+	}
+	// Initialize the locker if it is not already set.
+	if c.locker == nil {
+		home, err := os.UserHomeDir()
+		if err != nil {
+			return fmt.Errorf("home: %w", err)
+		}
+
+		c.locker, err = newLocker(filepath.Join(home, lockFilePath))
+		if err != nil {
+			return fmt.Errorf("locker: %w", err)
+		}
+	}
+	return nil
+}
+
+// load loads the token cache file from disk. If the file is corrupt or if its
+// version does not match tokenCacheVersion, it returns an error.
+func (c *FileTokenCache) load() (*tokenCacheFile, error) {
+	raw, err := os.ReadFile(c.fileLocation)
+	if err != nil {
+		return nil, fmt.Errorf("read: %w", err)
+	}
+	f := &tokenCacheFile{}
+	err = json.Unmarshal(raw, &f)
+	if err != nil {
```

Comment:
[nit] This is more idiomatic and slightly more robust as we do not expect this error to "exist" outside the if statement. 

```suggestion
	if err := json.Unmarshal(raw, &f); err != nil {
```